test/integration/controlplane/kube_apiserver_test.go

```diff
@@ -606,3 +608,71 @@ func TestMultiAPIServerNodePortAllocation(t *testing.T) {
 	}
 
 }
+
+func TestKubernetesDefaultServiceReconcile(t *testing.T) {
+	serviceCIDR := "192.168.0.0/24"
+
+	client, _, tearDownFn := framework.StartTestServer(t, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			opts.ServiceClusterIPRanges = serviceCIDR
+		},
+	})
+	defer tearDownFn()
+
+	var defaultService *corev1.Service
+	// Wait until the default "kubernetes" service is created.
+	if err := wait.Poll(250*time.Millisecond, time.Minute, func() (bool, error) {
+		var err error
+		defaultService, err = client.CoreV1().Services(metav1.NamespaceDefault).Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		return !apierrors.IsNotFound(err), nil
+	}); err != nil {
+		t.Fatalf("creating kubernetes service timed out")
+	}
+
+	// update the Service
+	svc := defaultService.DeepCopy()
+	svc.Spec.Ports = append(svc.Spec.Ports, corev1.ServicePort{
+		Name:       "wrong-port",
+		Port:       80,
+		TargetPort: intstr.IntOrString{IntVal: 8080},
+	})
+	_, err := client.CoreV1().Services(metav1.NamespaceDefault).Update(context.TODO(), svc, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	// validate the service was not reconciled
+	if err := wait.Poll(250*time.Millisecond, time.Minute, func() (bool, error) {
+		svc, err := client.CoreV1().Services(metav1.NamespaceDefault).Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		return !reflect.DeepEqual(svc.Spec, defaultService.Spec), nil
+	}); err != nil {
+		t.Fatalf("reconciling kubernetes service timed out")
```

Comment:
I'm not following this - shouldn't it actually fail?

Reply:
if the services are equal it returns successfully